lib/shopify_cli/tasks/update_dashboard_urls.rb

```diff
@@ -25,19 +26,28 @@ def call(ctx, url:, callback_url:)
         raise
       end
 
-      def construct_redirect_urls(urls, new_url, callback_url)
+      def construct_redirect_urls(urls, new_url, callback_urls)
         new_urls = urls.map do |url|
           if (match = url.match(NGROK_REGEX))
             "#{new_url}#{match[2]}"
           else
             url
           end
         end
-        if new_urls.grep(/#{new_url}#{callback_url}/).empty?
-          new_urls.push("#{new_url}#{callback_url}")
+        callback_urls.each do |callback_url|
+          if new_urls.grep(/#{new_url}#{callback_url}/).empty?
+            new_urls.push("#{new_url}#{callback_url}")
+          end
         end
         new_urls.uniq
       end
+
+      private
+
+      def already_updated(app, constructed_urls, url)
+        app["applicationUrl"].match(url) &&
+          Set.new(constructed_urls) == Set.new(app["redirectUrlWhitelist"])
```

Comment:
What about renaming the parameters to make it easier to understand?

```suggestion
      def already_updated(app, new_redirect_urls, new_url)
        current_url = app["applicationUrl"]
        current_redirect_urls = app["redirectUrlWhitelist"]
        current_url.match(new_url) &&
          Set.new(current_redirect_urls) == Set.new(new_redirect_urls)
```